test/CowEvcOpenPositionWrapper.t.sol

```diff
@@ -0,0 +1,661 @@
+    function test_OpenPositionWrapper_ValidateWrapperData() external view {
+        address account = address(uint160(user) ^ 1);
+        CowEvcOpenPositionWrapper.OpenPositionParams memory params = _createDefaultParams(user, account);
+
+        bytes memory wrapperData = abi.encode(params, new bytes(0));
+
+        // Should not revert for valid wrapper data
+        openPositionWrapper.validateWrapperData(wrapperData);
+    }
+
+    /// @notice Test setting pre-approved hash
+    function test_OpenPositionWrapper_SetPreApprovedHash() external {
+        vm.skip(bytes(forkRpcUrl).length == 0);
+
+        address account = address(uint160(user) ^ 1);
+        CowEvcOpenPositionWrapper.OpenPositionParams memory params = _createDefaultParams(user, account);
+        bytes32 hash = openPositionWrapper.getApprovalHash(params);
+
+        // Initially hash should not be approved
+        assertEq(openPositionWrapper.preApprovedHashes(user, hash), 0, "Hash should not be approved initially");
+
+        // User pre-approves the hash
+        vm.prank(user);
+        vm.expectEmit(true, true, false, true);
+        emit PreApprovedHashes.PreApprovedHash(user, hash, true);
+        openPositionWrapper.setPreApprovedHash(hash, true);
+
+        // Hash should now be approved
+        assertGt(openPositionWrapper.preApprovedHashes(user, hash), 0, "Hash should be approved");
+
+        // User revokes the approval
+        vm.prank(user);
+        vm.expectEmit(true, true, false, true);
+        emit PreApprovedHashes.PreApprovedHash(user, hash, false);
+        openPositionWrapper.setPreApprovedHash(hash, false);
+
+        // Hash should no longer be approved
+        assertEq(
+            openPositionWrapper.preApprovedHashes(user, hash),
+            uint256(keccak256("PreApprovedHashes.Consumed")),
+            "Hash should not be approved after revocation"
+        );
+    }
+
+    /// @notice Test opening a position with pre-approved hash (no signature needed)
+    function test_OpenPositionWrapper_WithPreApprovedHash() external {
+        vm.skip(bytes(forkRpcUrl).length == 0);
+
+        address account = address(uint160(user) ^ 1);
+
+        // Create params using helper
+        CowEvcOpenPositionWrapper.OpenPositionParams memory params = _createDefaultParams(user, account);
+
+        // Get settlement data
+        SettlementData memory settlement = getOpenPositionSettlement({
+            owner: user,
+            receiver: account,
+            sellToken: WETH,
+            buyVaultToken: ESUSDS,
+            sellAmount: DEFAULT_BORROW_AMOUNT,
+            buyAmount: DEFAULT_BUY_AMOUNT
+        });
+
+        // Setup user approvals and pre-approve hash
+        bytes32 hash = openPositionWrapper.getApprovalHash(params);
+        _setupUserPreApprovedFlow(account, hash);
+
+        // User pre-approves the order on CowSwap
+        // Does not need to run here because its executed in `setupCowOrder`
```

Comment:
This function doesn't exist anymore. (Same suggestion in the test with the other authentication path.)

```suggestion
        // Does not need to run here because it was signed as part of the settlement creation
```